database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -24,8 +24,8 @@
 /**
  * This class implements an array-like collection on top of a Firebase location.
  */
-class FirebaseArray implements ChildEventListener {
-    public interface OnChangedListener {
+public class FirebaseArray implements ChildEventListener {
```

Comment:
Oh and I forgot to mention: abstraction from `ArrayList<DataSnapshot> mSnapshots` is still maintained with this new model.